src/peft/tuners/lora/model.py

```diff
@@ -377,6 +377,15 @@ def _unload_and_optionally_merge(
             if getattr(self.model, "quantization_method", None) == "gptq":
                 raise ValueError("Cannot merge LORA layers when the model is gptq quantized")
 
+        adapters_to_consider = adapter_names or self.active_adapters
+        is_modules_to_save_available = any(
+            self.peft_config[adapter].modules_to_save is not None for adapter in adapters_to_consider
+        )
+        if is_modules_to_save_available and len(adapters_to_consider) > 1:
```

Comment:
If `adapter_names` is a str, this will trigger an error even though it would be fine, right?

Reply:
`adapter_names` is a `List[str]`, right?